lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -852,27 +846,35 @@ export class UserAgentApplication {
         this.cacheStorage.setItem(Constants.renewStatus + expectedState, Constants.tokenRenewStatusCancelled);
       }
     }, this.config.system.loadFrameTimeout);
+
+    const iframe = await this.loadFrame(urlNavigate, frameName);
+    const hash = await this.pollWindowForHash(iframe.contentWindow);
+    this.handleAuthenticationResponse(hash);
   }
 
   /**
    * @hidden
    * Loads iframe with authorization endpoint URL
    * @ignore
    */
-  private loadFrame(urlNavigate: string, frameName: string): void {
+  private loadFrame(urlNavigate: string, frameName: string): Promise<HTMLIFrameElement> {
     // This trick overcomes iframe navigation in IE
     // IE does not load the page consistently in iframe
     this.logger.info("LoadFrame: " + frameName);
     const frameCheck = frameName;
 
-    setTimeout(() => {
-      const frameHandle = this.addHiddenIFrame(frameCheck);
-      if (frameHandle.src === "" || frameHandle.src === "about:blank") {
-        frameHandle.src = urlNavigate;
-        this.logger.infoPii("Frame Name : " + frameName + " Navigated to: " + urlNavigate);
-      }
-    },
-    this.config.system.navigateFrameWait);
+    return new Promise((resolve) => {
```

Comment:
never reject?

Reply:
I couldn't think of an obvious scenario where this would happen, but I'll definitely add something just in case.